src/optimizer/column_lifetime_analyzer.cpp

```diff
@@ -62,8 +62,8 @@ void ColumnLifetimeAnalyzer::VisitOperator(LogicalOperator &op) {
 			break;
 		}
 		auto &comp_join = op.Cast<LogicalComparisonJoin>();
-		if (comp_join.join_type == JoinType::MARK || comp_join.join_type == JoinType::SEMI ||
-		    comp_join.join_type == JoinType::ANTI) {
+		if (comp_join.join_type == JoinType::MARK || comp_join.join_type == JoinType::LEFT_SEMI ||
+		    comp_join.join_type == JoinType::LEFT_ANTI) {
 			break;
```

Comment:
I think this is working correctly, but can we add a test where not all RHS values are propagated and verify that the optimizer projects them out?

For example:
```sql
CREATE TABLE lhs AS SELECT range % 5 i, range j FROM range(100);
CREATE TABLE rhs AS SELECT range % 5 i, range j FROM range(10);
-- JoinOrderOptimizer should flip this to a RIGHT_SEMI
SELECT rhs.j FROM rhs SEMI JOIN lhs USING (i);
-- The plan should not contain a PROJECTION,
-- because i was already projected out by the ColumnLifetimeAnalyzer
```

Reply:
doesn't `j` still need to be projection out of the right semi join though? The has join has columns `rhs.j`, `lhs.i`, `rhs.i`. Afterwards just `rhs.j` is projected out.
```
┌───────────────────────────┐
│      EXPLAIN_ANALYZE      │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│             0             │
│          (0.00s)          │
└─────────────┬─────────────┘
┌─────────────┴─────────────┐
│         PROJECTION        │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│             j             │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│             10            │
│          (0.00s)          │
└─────────────┬─────────────┘
┌─────────────┴─────────────┐
│         HASH_JOIN         │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│         RIGHT SEMI        │
│           i = i           │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ├──────────────┐
│           EC: 10          │              │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │              │
│             10            │              │
│          (0.00s)          │              │
└─────────────┬─────────────┘              │
┌─────────────┴─────────────┐┌─────────────┴─────────────┐
│         SEQ_SCAN          ││         SEQ_SCAN          │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│            lhs            ││            rhs            │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│             i             ││             i             │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││             j             │
│          EC: 100          ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││           EC: 10          │
│            100            ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│          (0.00s)          ││             10            │
│                           ││          (0.00s)          │
└───────────────────────────┘└───────────────────────────┘
```
This is what I'm seeing is all

Reply:
Right, never mind. I just checked how the projection map is used, and it will still always have a projection on top. This is something we might want to look at in the future, but for now if the query is successful I'd say it's good